examples/hand_pose_estimation/backend_keypoints.py

```diff
@@ -0,0 +1,1163 @@
+        keypoints2D, keypoints2D_visibility, image_size, crop_size)
+    scale, scale_matrix = get_scale_matrix(scale)
+    image_crop = crop_image_from_coordinates(
+        image, crop_center, crop_size, scale)
+    keypoint_uv21 = get_keypoints_camera_coordinates(
+        keypoints2D, crop_center, scale, crop_size)
+    scale_translation_matrix = get_scale_translation_matrix(
+        crop_center, crop_size, scale)
+    scale_matrix_uv = np.matmul(scale_matrix, camera_matrix)
+    camera_matrix_cropped = np.matmul(scale_translation_matrix, scale_matrix_uv)
+    return scale, np.squeeze(image_crop), keypoint_uv21, camera_matrix_cropped
+
+
+def flip_right_hand(canonical_keypoints3D, flip_right):
+    """ Flip right hend to left hand coordinates.
+
+    # Arguments
+        canonical_keypoints3D: Numpy array of shape (21, 3).
+        flip_right: boolean value.
+
+    # Returns
+        canonical_keypoints3D_left: Numpy array of shape (21, 3).
+    """
+    shape = canonical_keypoints3D.shape
+    expanded = False
+    if len(shape) == 2:
+        canonical_keypoints3D = np.expand_dims(canonical_keypoints3D, 0)
+        flip_right = np.expand_dims(flip_right, 0)
+        expanded = True
+    canonical_keypoints3D_mirrored = np.stack(
+        [canonical_keypoints3D[:, :, 0], canonical_keypoints3D[:, :, 1],
+         -canonical_keypoints3D[:, :, 2]], -1)
+    canonical_keypoints3D_left = np.where(
+        flip_right, canonical_keypoints3D_mirrored, canonical_keypoints3D)
+    if expanded:
+        canonical_keypoints3D_left = np.squeeze(canonical_keypoints3D_left,
+                                                axis=0)
+    return canonical_keypoints3D_left
+
+
+def extract_dominant_hand_visibility(keypoint_visibility, dominant_hand):
+    """ Extract Visibility mask for dominant hand.
+
+    # Arguments
+        keypoint_visibility: Numpy array of shape (21, 1).
+        dominant_hand: List of size (2).
+
+    # Returns
+        keypoint_visibility_21: Numpy array of shape (21, 2).
+    """
+    keypoint_visibility_left = keypoint_visibility[:LEFT_LAST_KEYPOINT_ID + 1]
+    keypoint_visibility_right = keypoint_visibility[RIGHT_ROOT_KEYPOINT_ID:
+                                                    RIGHT_LAST_KEYPOINT_ID + 1]
+    keypoint_visibility_21 = np.where(
+        dominant_hand[:, 0], keypoint_visibility_left,
+        keypoint_visibility_right)
+    return keypoint_visibility_21
+
+
+def extract_dominant_keypoints2D(keypoint_2D, dominant_hand):
+    """ Extract keypoint 2D.
+
+    # Arguments
+        keypoint_2D: Numpy array of shape (21, 2).
+        dominant_hand: List of size (2) with booleans.
+
+    # Returns
+        keypoint_visibility_2D_21: Numpy array of shape (21, 2).
+    """
+    keypoint_visibility_left = keypoint_2D[:LEFT_LAST_KEYPOINT_ID + 1, :]
+
+    keypoint_visibility_right = keypoint_2D[RIGHT_ROOT_KEYPOINT_ID:
+                                            RIGHT_LAST_KEYPOINT_ID + 1, :]
```

Comment:
Why if it's the last one do you need to sum a + 1? If you are always using the *_LAST_KEYPOINT_ID + 1 maybe we should redifine what we mean by the last keypoint id